### PR TITLE
CVE-2013-1665

### PR DESCRIPTION
**Pull request by dungtran3864**
CVE-2013-1665

### CODE REVIEW

File: cves/CVE-2013-1665.yml [hunk 1]
```diff
@@ -49,7 +49,17 @@ description_instructions: |
...
+  The vulnerability exists in the way XML parsers and libraries used by Django
+  handle XML deserialization.
+  The implementation, which takes an XML and constructs into an object from the
+  structured data, does not handle parsing incorrectly. The DOCTYPE element
+  of XML can be injected with dangerously customized entities that are loaded
+  outside of the application.
+  Attackers can take advantage of this vulnerability by injecting network requests
+  via entity declaration in the DOCTYPE element of XML, forwarding and rebouding
+  with the server, and finally gaining read access to files. The server may also
+  take long time to process those requests, resulting in a DoS attack.
```
Comment (rxs7498) [inline cves/CVE-2013-1665.yml]: This is a well explained and detailed description of the vulnerability. There are a couple of grammar errors.

File: cves/CVE-2013-1665.yml [hunk 2]
```diff
@@ -146,8 +164,14 @@ autodiscoverable:
...
-  answer_note:
-  answer:
+  answer_note: |
+    This is an XML external entity attack. Python documentation even states that
+    the issue is nothing new. Billion laughs, an XML external entity attack,
+    has been reported since 2003. It's plausible that an automated tool
+    could have been developed after several such similar reports to discover these
+    security loopholes. The tool can also get its forbidden entity libraries updated
+    overtime for more rigorous vulnerability discoveries in the long run.
+  answer: true
```
Comment (rxs7498) [inline cves/CVE-2013-1665.yml]: Be sure to implement sources

File: cves/CVE-2013-1665.yml [hunk 3]
```diff
@@ -159,8 +183,11 @@ specification:
 
     The answer field should be boolean. In answer_note, please explain
     why you come to that conclusion.
-  answer_note:
-  answer:
+  answer_note: |
+    A blog from Python on this vulnerability indicates that the developers have
+    to re-implement the XML specifications properly, which indicates that there's
+    a violation of XML specifications.
+  answer: true
```
Comment (rxs7498) [inline cves/CVE-2013-1665.yml]: Make sure to include the link to said blog, or and add maybe quotes to it. Suggest using APA format for references

File: cves/CVE-2013-1665.yml [hunk 4]
```diff
@@ -66,30 +76,34 @@ fixes_vcc_instructions: |
...
-  note: This VCC was discovered automatically via archeogit.
+  note: The developer is adding more performance tests for serializers.
 - commit: 7019123d21f202bfce54ad30eb33ae9f4b3bcef9
-  note: This VCC was discovered automatically via archeogit.
+  note: The developer is fixing an exception handling with deserializers.
 - commit: ac64e91a0cadc57f4bc5cd5d66955832320ca7a1
-  note: This VCC was discovered automatically via archeogit.
+  note: The developer is merging branches.
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
```
Comment (rxs7498) [inline cves/CVE-2013-1665.yml:108]: Upvote from me

File: cves/CVE-2013-1665.yml [hunk 5]
```diff
@@ -287,7 +344,7 @@ CWE_instructions: |
 
   Just the number here is fine. No need for name or CWE prefix. If more than one
   apply here, then choose the best one and mention the others in CWE_note.
-CWE:
+CWE: 200
```
Comment (jbz6587) [inline cves/CVE-2013-1665.yml:349]: Upvote +2

Comment (jbz6587) [reply cves/CVE-2013-1665.yml:349]: Overall very good CVE research, very detailed and well articulated. Although some grammar mistakes are present and have been previously pointed out.

File: cves/CVE-2013-1665.yml [hunk 6]
```diff
@@ -239,17 +278,26 @@ lessons:
...
     applies:
     note:
   serial_killer:
-    applies:
-    note:
+    applies: true
+    note: |
+      The vulnerability involves deserialization of XML, which could result in
+      potential confidentiality disclosure if it's not handled correctly. Also,
+      serialization is not encryption, so it's good to practice more defensive
```
Comment (CalvinWu4) [inline cves/CVE-2013-1665.yml:301]: Great point about serialization/encryption

File: cves/CVE-2013-1665.yml [hunk 7]
```diff
@@ -49,7 +49,17 @@ description_instructions: |
 
   Your target audience is people just like you before you took any course in
   security
-description:
+description: |
+  The vulnerability exists in the way XML parsers and libraries used by Django
+  handle XML deserialization.
+  The implementation, which takes an XML and constructs into an object from the
+  structured data, does not handle parsing incorrectly. The DOCTYPE element
```
Comment (CalvinWu4) [inline cves/CVE-2013-1665.yml]: You mean "parsing correctly"

File: cves/CVE-2013-1665.yml [hunk 8]
```diff
@@ -49,7 +49,17 @@ description_instructions: |
 
   Your target audience is people just like you before you took any course in
   security
-description:
+description: |
```
Comment (CalvinWu4) [inline cves/CVE-2013-1665.yml:52]: Great explanation of why deserialization is not secure enough

File: cves/CVE-2013-1665.yml (hunk 1 above)
Comment (dungtran3864) [reply cves/CVE-2013-1665.yml]: Thank you. Will fix the grammar. Might be even more helpful if you can pin point the grammar mistakes.

File: cves/CVE-2013-1665.yml (hunk 2 above)
Comment (dungtran3864) [reply cves/CVE-2013-1665.yml]: I will put on the website that I look at here.

File: cves/CVE-2013-1665.yml (hunk 3 above)
Comment (dungtran3864) [reply cves/CVE-2013-1665.yml]: Yup, I will put in the link of the blog

File: cves/CVE-2013-1665.yml (hunk 4 above)
Comment (dungtran3864) [reply cves/CVE-2013-1665.yml:108]: Thanks!

File: cves/CVE-2013-1665.yml (hunk 5 above)
Comment (dungtran3864) [reply cves/CVE-2013-1665.yml:349]: Thank you! I will correct those mistakes.

File: cves/CVE-2013-1665.yml (hunk 6 above)
Comment (dungtran3864) [reply cves/CVE-2013-1665.yml:301]: Thank you!

File: cves/CVE-2013-1665.yml (hunk 7 above)
Comment (dungtran3864) [reply cves/CVE-2013-1665.yml]: Oh, yup. It's a typo mistake..

File: cves/CVE-2013-1665.yml (hunk 8 above)
Comment (dungtran3864) [reply cves/CVE-2013-1665.yml:52]: Thank you!

**Pull request merged**